docs/libraries/nodejs/examples.md

```diff
@@ -1,101 +1,148 @@
+* `account.balance()`: returns the balance for the given account
+* `account.sync()`: sync the account information with the tangle
+
+## Generating address(es)
+Each account can posses multiple addresses. Addresses are generated deterministically based on the account and address index. It means that the combination of account and index uniquely identifies the given address.
+
+Addresses are of two types: `internal` and `public` (external):
+* each set of addresses is independent from each other and has independent `index` id
+* addresses that are created by `account.generateAddress()` are indicated as `internal=false` (public)
+* internal addresses (`internal=true`) are so called `change` addresses and are used to send the excess funds to
+* the approach is also known as a *BIP32 Hierarchical Deterministic wallet (HD Wallet)*.
+
+_Note: You may remember IOTA 1.0 network in which addresses were not reusable. It is no longer true and addresses can be reused multiple times in IOTA 1.5 (Chrysalis) network._
 
-Code:
 ```javascript
 {{ #include ../../../bindings/nodejs/examples/2-generate-address.js }}
 ```
 
-## 3. Example: Check Balance
-Run:
-```
-node 3-check_balance
-```
+## Checking balance
+Before we continue further, go to [IOTA testnet faucet service](https://faucet.testnet.chrysalis2.com/) and send to your testnet addresses some tokens.
 
-Code:
 ```javascript
 {{ #include ../../../bindings/nodejs/examples/3-check_balance.js }}
 ```
 
-## 4. Example: Check Balance
-Now you can send the test tokens to an address! 
-
-Run
-```
-node 4-send.js
-```
+## Sending tokens
+Sending tokens is performed via `Account` instance:
 
-Code:
 ```javascript
 {{ #include ../../../bindings/nodejs/examples/4-send.js }}
 ```
+The full function signature is `Account.send(address, amount[, options])`.
+Default options are perfectly fine and do the job done, however additional options can be provided, such as `remainderValueStrategy`:
+* `changeAddress`: Send the remainder value to an internal address
+* `reuseAddress`: Send the remainder value back to its original address
 
-## 5. Backup
+`Account.send()` function returns a `wallet message` that fully describes the given transaction. Especially `messageId` can later be used for checking a confirmation status. Individual messages related to the given account can be obtained via `Account.listMessages()` function.
 
-Run
-```
-node 5-backup.js
-```
+### Dust protection
+Please note, there is also implemented a [dust protection](https://chrysalis.docs.iota.org/guides/dev_guide.html#dust-protection) mechanism in the network protocol to avoid malicious actors to spam network in order to decrease node performance while keeping track of unspent amount (`UTXO`):
+> "... microtransaction below 1Mi of IOTA tokens [can be sent] to another address if there is already at least 1Mi on that address"
+That's why we did send 1Mi in the given example to comply with the protection."
```

Comment:
Maybe we should also add an example where it would fail because of this, like if you have 4.5 Mi and wants to send 4 Mi, it will fails because you can't leave dust behind

Reply:
Good point. My plan is to review wallet.rs docs in different PR and will better integrate wallet.rs docs with client.rs docs and chrysalis docs. And will mention this in Dust protection as well.

Reply:
I targeted this point as a part of https://github.com/iotaledger/wallet.rs/pull/598